lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1213 @@
+            cmd += ['--ulimit', params['ulimit']]
+        if params.get('user'):
+            cmd += ['--user', params['user']]
+        if params.get('userns'):
+            cmd += ['--userns', params['userns']]
+        if params.get('uts'):
+            cmd += ['--uts', params['uts']]
+        if params.get('volume'):
+            for vol in params['volume']:
+                cmd += ['--volume', vol]
+        if params.get('volumes_from'):
+            for vol in params['volumes_from']:
+                cmd += ['--volumes-from', vol]
+        if params.get('workdir'):
+            cmd += ['--workdir', params['workdir']]
+        # Add your own args for podman command
+        if params.get('cmd_args'):
+            cmd += params['cmd_args']
+        cmd.append(params['image'])
+        if params.get('command'):
+            cmd.append(params['command'])
+        return cmd
+
+
+def ensure_image_exists(module, image):
+    rc, out, err = run_podman_command(module, args=['image', 'exists', image], ignore_errors=True)
+    if rc != 0:
+        rc, out, err = run_podman_command(module, args=['image', 'pull', image])
+        if rc != 0:
+            module.fail_json(msg="Can't pull image %s:\n%s\v%s" % (image, out, err))
+
+
+class PodmanContainer(object):
+    def __init__(self, module, name, params):
+
+        super(PodmanContainer, self).__init__()
+        self.module = module
+        self.name = name
+        self.info = self.get_info()
+
+    @property
+    def exists(self):
+        return self.info != {}
+
+    @property
+    def different(self):
+        # TODO(sshnaidm): implement difference calculation between input vars
+        # and current container to understand if we need to recreate it
+        return True
+
+    @property
+    def running(self):
+        return False if not self.exists else self.info['State']['Running']
+
+    @property
+    def stopped(self):
+        return False if not self.exists else not self.info['State']['Running']
+
+    def get_info(self):
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container', 'inspect', self.name], ignore_errors=True)
+        if rc != 0:
+            rc, out, err = run_podman_command(
+                module=self.module,
+                args=['container', 'exists', self.name], ignore_errors=True)
+            if rc == 0:
+                self.module.fail_json(msg="Can't inspect existing container %s" % self.name)
+            else:
+                return {}
+        else:
+            return json.loads(out)[0]
+
+    def _action(self, action):
+        command = construct_command_from_params(action, self.module.params, self.module)
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container'] + command,
+            ignore_errors=True)
+        if rc != 0:
+            self.module.fail_json(
+                msg="Can't %s container %s: %s" % (action, self.name, err))
```

Comment:
Don't render the whole output into a message. Use separate keys for those.
```suggestion
                msg="Can't %s container %s" % (action, self.name),
                stdout=out,
                stderr=err,
            )
```
These will automatically generate a few extra return keys (`stdout_lines` and `stderr_lines`).

Reply:
Didn't know this, thanks!